compiler/rustc_query_impl/src/plumbing.rs

```diff
@@ -693,7 +695,7 @@ macro_rules! define_queries {
                 }
             }
 
-            pub(crate) fn try_collect_active_jobs<'tcx>(tcx: TyCtxt<'tcx>, qmap: &mut QueryMap) {
+            pub(crate) fn try_collect_active_jobs<'tcx>(tcx: TyCtxt<'tcx>, qmap: &mut QueryMap, complete: &mut bool) {
```

Comment:
Can we just return `bool` here? I'd like to avoid having a function rely on extra mutable references.

Reply:
That adds more complexity though.